x-pack/platform/plugins/shared/streams/server/lib/streams/state_management/wired_stream.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  StreamDefinition,
+  WiredStreamDefinition,
+  isWiredStreamDefinition,
+} from '@kbn/streams-schema';
+import { cloneDeep } from 'lodash';
+import { IScopedClusterClient, Logger } from '@kbn/core/server';
+import { State } from './state';
+import {
+  StreamActiveRecord,
+  StreamChangeStatus,
+  StreamCommitStatus,
+  ValidationResult,
+} from './types';
+import { StreamsStorageClient } from '../service';
+import { syncWiredStreamDefinitionObjects } from '../helpers/sync';
+import { deleteStreamObjects } from '../stream_crud';
+
+export class WiredStream implements StreamActiveRecord {
+  definition: WiredStreamDefinition;
+  changeStatus: StreamChangeStatus;
+  commitStatus: StreamCommitStatus;
+
+  constructor(definition: WiredStreamDefinition) {
+    // What about the assets?
+    this.definition = definition;
+    this.changeStatus = 'unchanged';
+    this.commitStatus = 'uncomitted';
+  }
+
+  clone(): StreamActiveRecord {
+    return new WiredStream(cloneDeep(this.definition));
+  }
+
+  markForDeletion(): void {
+    this.changeStatus = 'deleted';
+  }
+
+  update(newDefinition: StreamDefinition) {
```

Comment:
We accept a StreamDefinition here because it could be that a WiredStream gets a change to a GroupStream which we would want to validate against and stop.

I haven't included the desired and current state parameters here but this function can apply changes to more streams in the state than only itself.